packages/client/src/OsdkObjectFrom.ts

```diff
@@ -32,44 +32,52 @@ type OsdkCommonFrom<
   Q extends ObjectTypeDefinition<any> | InterfaceDefinition<any, any>,
   L extends ObjectOrInterfacePropertyKeysFrom2<Q> =
     ObjectOrInterfacePropertyKeysFrom2<Q>,
+  R extends boolean = false,
 > =
   & {
     [P in L]: OsdkObjectPropertyType<
       Q["properties"][P]
     >;
   }
   & {
+    /** @deprecated use $apiName */
     __apiName: Q["apiName"];
+
+    $apiName: Q["apiName"];
+
+    /** @deprecated use $primaryKey */
     __primaryKey: Q extends ObjectTypeDefinition<any>
       ? OsdkObjectPrimaryKeyType<Q>
       : unknown;
-    // $uniqueId: string; // will be dynamic
 
-    /**
-     * Future versions will require explicitly asking for this field. For now we are marking
-     * as always optional to avoid breaking changes.
-     */
-    __rid?: string;
-  };
+    $primaryKey: Q extends ObjectTypeDefinition<any>
+      ? OsdkObjectPrimaryKeyType<Q>
+      : unknown;
+    // $uniqueId: string; // will be dynamic
+  }
+  & (R extends true ? { $rid: string } : {});
```

Comment:
can you use `{}` safely here? or do you want `Record<string, never>`?

Reply:
It doesn't matter in this case because after they are joined it has the other types.